x/metadata/keeper/keeper.go

```diff
@@ -182,3 +196,24 @@ func (k Keeper) ValidatePartiesInvolved(parties []types.Party, requiredParties [
 	}
 	return nil
 }
+
+// VerifyCorrectOwner to determines whether the signer resolves to the owner of the OSLocator record.
+func (k Keeper) VerifyCorrectOwner(ctx sdk.Context, ownerAddr sdk.AccAddress) bool { // nolint:interfacer
```

Comment:
This is one thing I was going to ask about. How does provenance know I'm only pushing public keys/addresses that I have the associated private key for? By reading the doc line it seems like this function is checking if the tx was signed by the address present in the request proto? Right now a p8e instance has one provenance account and signs and pays fees on behalf of possibly many internal key pairs. Its these public keys that are present on scopes and not the provenance account that's signing and sending Txs.

Reply:
For this specific endpoint the answer is straight forward.  The request must be signed by the address the record is for.  A notary could not sign for this transaction.  There is a need to review the larger signing approach for the metadata module though.  

All of the APIs added have used the standard signing methods supported by the SDK (not the independent signature/blockchain notary pattern).  This was done because there are actually several different features that are available for blockchain signatures that overlap with our needs and this module should strive to implement/use blockchain native approaches.  Some of these include module/smart contract accounts, multisig, fee payer/delegation, etc.  The P8e specific memorialize contract is the only endpoint that supports current p8e notary/independent account signature method.

Reply:
That makes sense. How does that look from an implementation perspective? Say there's a p8e key pairs {A, B, C} that are already configured and exist. A user goes into the p8e ui and creates a new key pair. The key pair is saved in p8e and object-store now we need to submit this request to provenance. This new keypair D will sign the tx and add it's public key as the address in the request.

Does provenance complain about that request? Is anything needed to first establish keypair D as a provenance account? Where does the fees get paid from?

Reply:
@scirner22 since the keypair(ne one) is assumed to be a valid address/account on chain, i mean this is moot now since the keypair has to pay fee's for the tx so cannot not exist to establish an object locator record, but pointing it out since there is a concept of delegated fee payer coming in the future(afaik)..having said that i can take out the check if that is the consensus 🤷

Reply:
We really really need the `authz` module from upstream that supports the "feepayer" concept to make a lot of the Figure use cases much easier to deal with.  Right now we have to do a bunch of fund shuffling to ensure the accounts that need to sign have the coin to pay the fees.

Reply:
I think this check is what we want because without it we would need some other type of check to prove your submitting public keys that you own the private key for, but I'm trying to understand what prereq steps we will need to take for every new keypair now.
Is there an endpoint to hit to establish a new address/account on chain? It does seem like what you mentioned is exactly what p8e would want though. The ability for accounts to sign transactions but have the sole figure p8e account to fund the fee.

Reply:
(note: its actually the feegrant module we want for this example, the authz module would let one account sign for another which I guess also applies here).

Reply:
right now with the state of current code base, fee's 
* would get paid from the signer
i.e the keypair that you established(and the current user setting up their locator)
 when you establish the keypair in an app/ui, transfer them hash to pay fees(from probably figure tech or provenance accnt, but not sure on that), 
that would
1. establish their account on chain
2. give them fees to establish their os locator record.

Reply:
I think this will be the rough flow even with fee payer.. with the payer option though instead of sending the new account hash the money account would create a grant allowing the target account to spend their funds for fees.

Reply:
Awesome. Thanks for working on this @arnabmitra . I'll probably reach out when I'm ready to start trying this. I'm probably a full week away from that point.